pages/api/dpu/setup.ts

```diff
@@ -11,6 +9,17 @@ const setupHandler = async (req: NextApiRequest, res: NextApiResponse) => {
 		res.status(400).json({ "error": "Invalid request body" });
 		return;
 	}
+	// delete previous installations
+	let installsRemoved = true;
+	await removePreviousInstallations(jsonBody.installationId).catch(err => {
+		console.error(`[setupHandler] Unable to remove previous installations for ${jsonBody.installationId}`, err);
+		res.status(500).json({"error": "Internal Server Error"});
+		installsRemoved = false;
+		return;
+	});
+	if (!installsRemoved) {
+		return;
+	}
```

Comment:
Is this actually working? The `res.status` command is inside the catch block (not the scope of the handler function) and I am not very confident that line 20 will consistently work.

If you awaiting, then why not directly use a `try-catch` block? That way, you will not even need the `installsRemoved` variable. If the async function fails, you can directly return the 500 response.